packages/botonic-react/src/webchat/messages-reducer.js

```diff
@@ -7,14 +7,30 @@ import {
   UPDATE_REPLIES,
 } from './actions'
 
+const getMsgComponentId = m => {
+  if (m.props.id) return m.props.id
+  if (m.props.value) return m.props.value.input && m.props.value.input.id
+  return undefined
+}
+
 export const messagesReducer = (state, action) => {
   switch (action.type) {
     case ADD_MESSAGE:
       return addMessageReducer(state, action)
     case ADD_MESSAGE_COMPONENT:
+      if (
+        state.messagesComponents &&
+        state.messagesComponents.find(
+          m => getMsgComponentId(m) === action.payload.id
+        )
+      )
+        return state
       return {
         ...state,
-        messagesComponents: [...state.messagesComponents, action.payload],
+        messagesComponents: [
+          ...(state.messagesComponents || []),
+          action.payload,
```

Comment:
Any risk for this to be `null | undefined` and have a parasite within this new array?

Reply:
not needed at all